### PR TITLE
chore: [IAI-25] Replaces biometric library to support more devices biometric options

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR replaces `react-native-touch-id` with `react-native-fingerprint-scanner` and updates all local implementations inside app codebase.

## How to test
Test on both physical and virtual devices to catch if any issue may occur, close the app and then try logging in through biometric recognition.

This implementation may solve the issue #3337

### CODE REVIEW

**Review by fabriziofff — CHANGES_REQUESTED** [review]
Trying the new library I found the following wrong behaviors:

### Android
Prerequisite: At least one biometric active on the device, `biometric recognition` active in app settings.
1. `Profile > Security > Biometric Recognition`, if the `Biometric Recognition` is active and I try to disable the toggle, after successfully passing the biometric check, the toggle remains active.
2. Initial identification screen > Biometric prompt > Cancel, the screen shows "Retry", as if there a wrong authentication.
3. Initial identification screen > Biometric prompt > Cancel > tap on the fingerprint icon > complete the biometric identification > The screen remains and the authentitcation doesn't continue.
Cashback
4. Wallet > Cashback Card > Quit Cashback > Biometric prompt > Cancel, it is not possible to press the numeric buttons of the pin 

### iOS
✅ on iPhone 8 (TouchID) and iPhone 11 (FaceID)

File: ts/sagas/startup/checkAcknowledgedFingerprintSaga.ts [hunk 1]
```diff
@@ -87,28 +87,66 @@ export function* checkAcknowledgedFingerprintSaga(): Generator<
...
+//               resolve("NOT_ENROLLED");
+//               break;
+//           }
+//         }
+//       })
+//       .catch(() => {
+//         resolve("UNAVAILABLE");
+//       });
+//   });
+// }
```
[the remark this answers is not in the document]
Comment (CrisTofani) [reply ts/sagas/startup/checkAcknowledgedFingerprintSaga.ts:152]: yes, my miss I left it for reference during implementation

File: package.json [hunk 2]
```diff
@@ -110,6 +110,7 @@
     "react-native-config": "1.4.3",
     "react-native-device-info": "^8.1.3",
     "react-native-exception-handler": "^2.10.8",
+    "react-native-fingerprint-scanner": "^6.0.0",
```
[the remark this answers is not in the document]
Comment (CrisTofani) [reply package.json:113]: e9bff9f solves

File: ts/sagas/startup/checkAcknowledgedFingerprintSaga.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (CrisTofani) [reply ts/sagas/startup/checkAcknowledgedFingerprintSaga.ts]: 659fe5a solves

**Review by fabriziofff — APPROVED** [review]

**Pull request merged**